### PR TITLE
[WASH-932] Add write file to iOS plugin

### PR DESCRIPTION
**Pull request by carrala**
- adds write file to iOS plugin and dart frontend
- also moves the group.leave() call in loginIdentity so that it matches the other methods (& verifies that it still works)

### CODE REVIEW

File: ios/Classes/SwiftFlutterPlugin.swift [hunk 1]
```diff
@@ -154,7 +158,32 @@ public class SwiftFlutterPlugin: NSObject, Flutter.FlutterPlugin {
         group.wait()
 
     }
-
+    
+    // MARK: WRITE FILE
```
Comment (efabens) [inline ios/Classes/SwiftFlutterPlugin.swift:162]: Is this comment still needed?

File: example/lib/main.dart [hunk 2]
```diff
@@ -95,6 +95,28 @@ class _MyAppState extends State<MyApp> {
...
+                                  publicSignKey: "",
+                                  privateSigningKey: "",
+                                  host: "",
+                                  email: "",
+                                  clientName: "");
+    var client = PluginTozny(creds);
+    try {
+      RecordMeta writtenFileMeta = await writeFile(client);
+      // verify the record was created & can be read
+      Record fileRecord = await readRecord(writtenFileMeta.recordID, client);
```
Comment (efabens) [inline example/lib/main.dart:113]: Is there a way to see that this file is actually in the file system?

**Review by galxy25 — APPROVED** [review]

**Review by sjonz — APPROVED** [review]

File: example/lib/main.dart [hunk 3]
```diff
@@ -95,6 +95,28 @@ class _MyAppState extends State<MyApp> {
...
+                                  publicKey: "",
+                                  privateKey: "",
+                                  publicSignKey: "",
+                                  privateSigningKey: "",
+                                  host: "",
+                                  email: "",
+                                  clientName: "");
+    var client = PluginTozny(creds);
+    try {
+      RecordMeta writtenFileMeta = await writeFile(client);
```
Comment (efabens) [inline example/lib/main.dart:111]: I feel like I am missing something. Does this need more parameters about record type and the file path to upload?

Comment (carrala) [reply example/lib/main.dart:111]: writeFile is already a method within main.dart (https://github.com/tozny/flutter_plugin/blob/wash-932-write-file-record/example/lib/main.dart#L202) & the record type & file path are defined there.

File: ios/Classes/SwiftFlutterPlugin.swift (hunk 1 above)
Comment (Mrhea) [reply ios/Classes/SwiftFlutterPlugin.swift:162]: It's more or less a convenience comment used in Xcode projects. A comment with MARK is seen in an Xcode file's sidebar (that shows the zoomed out view of the entire file) as large print. So you can more easily navigate to each method at a glance

**Review by Mrhea — APPROVED** [review]

File: example/lib/main.dart (hunk 2 above)
Comment (carrala) [reply example/lib/main.dart:113]: @efabens it turns out i will actually need to make a readFile method to actually download the file, so for now this is just verifying that the record actually exists. once i do that, i'll be able to check that the file is in the file system.

Comment (efabens) [reply example/lib/main.dart:113]: 👍🏻

**Pull request merged**